java/src/main/java/ai/onnxruntime/ONNXAllocator.java

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright © 2019, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime;
+
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.DoubleBuffer;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+import java.nio.ShortBuffer;
+
+/**
+ * An ONNX runtime memory allocator.
+ */
```

Comment:
Is it possible to have package only access in Java? Similar to 'internal' in C#? In that case I would make some of the classes, including this one, accessible from package only , instead of public.

In general, if there is no use case for a Java user to use the class, then, I would make it package only.

Reply:
Package only access is a valid concept in Java. However earlier conversations said that the allocator should be split out from the session. I had previously encapsulated it in the session. If there are user controlled allocators they need a public class.

Does the default allocator require synchronization? That's going to be a little tricky to do from Java with the default one.

Reply:
Done.